lab-jonathan/lib/authMiddleware.js

```diff
@@ -0,0 +1,39 @@
+let User = require('../model/model');
+let createError = require('http-errors');
+
+
+module.exports = (req, res, next) => {
+  let auth = req.headers.authorization;
+  if(!auth){
+    return next(createError(401, 'there is no auth'));
+  }
+  let base64String = auth.split('Basic ')[1];
+  let [username, password] = new Buffer(base64String, 'base64').toString().split(':');
+
+  User.findOne({username: username})
+  .then(user => {
+    let temp = user.hashPassword(password)
```

Comment:
hashing a new pw, or should this be comparing an existing hash?

The overall structure of this is off, since you're not using the right schema method. You want to get a user from the findOne method, then return either an error, if the user doesnt exist or isn't auth'd, or return comparePassword().